src/main/java/seedu/duke/parser/commandparser/UpdateParser.java

```diff
@@ -1,11 +1,215 @@
+                parsedAttributes[INDEX_OF_VENUE] = attribute[1];
+            }
+            if (update.contains(BUDGET_FLAG)) {
+                isFlag = true;
+                budget = convertEventBudgetToDouble(attribute[1]);
+            }
+            if (update.contains(DESCRIPTION_FLAG)) {
+                isFlag = true;
+                parsedAttributes[INDEX_OF_DESCRIPTION] = attribute[1];
+            }
+            if (update.contains(TASK_FLAG)) {
+                attribute[1] = attribute[1].replaceAll("\\s", "");
+                return parseUpdateTask(event, attribute[1]);
+            }
+            if (!isFlag) {
+                System.out.println("Invalid update, you have returned to the main page!");
+            }
+        }
+        return new UpdateEventCommand(event, parsedAttributes, dateTime, budget);
+    }
+
+    protected static Command parseUpdateTask(Event eventToBeUpdated, String index)
+            throws DukeException {
+        int taskNum = Integer.parseInt(index) - 1;
+        Task taskToBeUpdated = eventToBeUpdated.getFromTaskList(taskNum);
+        String[] userUpdates = prepareTaskUpdates(taskToBeUpdated);
+
+        String[] parsedAttributes = new String[3];
+        LocalDateTime dateTime = null;
+
+        for (String update : userUpdates) {
+            String[] attribute = update.trim().split("/+");
+            if (update.contains(TITLE_FLAG)) {
+                parsedAttributes[INDEX_OF_TITLE] = attribute[1];
+            } else if (update.contains(DEADLINE_FLAG)) {
+                dateTime = Parser.convertDateTime(attribute[1]);
+            } else if (update.contains(DESCRIPTION_FLAG)) {
+                parsedAttributes[INDEX_OF_DESCRIPTION] = attribute[1];
+            } else if (update.contains(MEMBER_FLAG)) {
+                attribute[1] = attribute[1].replaceAll("\\s", "");
+                return changeMember(attribute[1], taskToBeUpdated);
+            } else if (update.contains(REMOVE_FLAG)) {
+                attribute[1] = attribute[1].replaceAll("\\s", "");
+                return removeMember(attribute[1], taskToBeUpdated);
+            } else if (update.contains(ADD_FLAG)) {
+                return addMember(taskToBeUpdated);
+            } else {
+                System.out.println("Invalid update");
+            }
+        }
+        return new UpdateTaskCommand(parsedAttributes, taskToBeUpdated, dateTime);
+    }
+
+    private static Command changeMember(String index, Task taskToBeUpdated) throws DukeException {
+        int memberToBeReplaced = Integer.parseInt(index);
+        int memberIndex = prepareMemberDetails();
+        checkValidMember(memberIndex, taskToBeUpdated);
+        return new UpdateMemberCommand("change", memberIndex, taskToBeUpdated, memberToBeReplaced);
+    }
+
+    private static Command addMember(Task taskToBeUpdated) throws DukeException {
+        int memberIndex = prepareMemberDetails();
+        checkValidMember(memberIndex, taskToBeUpdated);
```

Comment:
Another check might need to be performed to consider the case if the member is already assigned to the task.

Reply:
makes sense, will do so!